### PR TITLE
XML BBE fix: continuing #21796

### PR DESCRIPTION
**Pull request by rdhananjaya**
## Purpose
Fix xml functions BBE and apply review suggestions of https://github.com/ballerina-platform/ballerina-lang/pull/21796

Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: examples/xml-access/xml_access.description [hunk 1]
```diff
@@ -1,2 +1,2 @@
 // XML elements can contain child XML items. Ballerina allows you to access these
-// child elements conveniently using xml step expressions and xml filter expressions.
+// child elements conveniently using xml step expressions and XML filter expressions.
```
Comment (praneesha) [inline examples/xml-access/xml_access.description]:
```suggestion
// child elements conveniently using XML step expressions and XML filter expressions.
```

File: examples/xml-attributes/xml_attributes.bal [hunk 2]
```diff
@@ -15,19 +15,19 @@ public function main() {
     string|error count = x1.count;
     io:println(count);
 
-    // Returns an error when accessing non existing attribute.
+    // Accessing a non-existent attribute will return an error.
     string|error count2 = x1.count2;
     io:println(count2 is error);
 
-    // It is possible to get all the attributes of a xml element.
+    // It is possible to get all the attributes of a `xml` element.
```
Comment (praneesha) [inline examples/xml-attributes/xml_attributes.bal]:
```suggestion
    // It is possible to get all the attributes of an `xml` element.
```

File: examples/xml-attributes/xml_attributes.bal [hunk 3]
```diff
@@ -15,19 +15,19 @@ public function main() {
...
     map<string> attributeMap = x1.getAttributes();
     io:println(attributeMap);
 
     // An attribute can also be accessed using the string representation of the qualified name from the attribute map.
     string? s = attributeMap["{http://ballerina.com/aa}status"];
     io:println(s);
 
-    // Updates an attribute by updating the attribute map of a xml element.
+    // Updates an attribute by updating the attribute map of a `xml` element.
```
Comment (praneesha) [inline examples/xml-attributes/xml_attributes.bal]:
```suggestion
    // Update an attribute by updating the attribute map of an `xml` element.
```

File: examples/xml-functions/xml_functions.bal [hunk 4]
```diff
@@ -1,27 +1,25 @@
...
-    io:println(bookName.getElementName());
-
-    // Gets the text content of an XML element.
-    io:println(bookName.getTextValue());
+    xmllib:Element bookNameElem = <xmllib:Element> bookName;
+    io:println(bookNameElem.getName());
 
-    // Checks if the XML element is empty.
-    io:println(content.isEmpty());
+    // Concatenates xml and string values.
```
Comment (praneesha) [inline examples/xml-functions/xml_functions.bal]:
```suggestion
    // Concatenate XML and string values.
```

File: examples/xml-functions/xml_functions.bal [hunk 5]
```diff
@@ -1,27 +1,25 @@
...
+    xml concat = xmllib:concat(someText, bookName, bookComment);
+    io:println(concat);
+    io:println(content == concat);
 
-    // Checks if the XML element has only one value.
-    io:println(content.isSingleton());
+    // Get the number of xml items in a sequence.
```
Comment (praneesha) [inline examples/xml-functions/xml_functions.bal]:
```suggestion
    // Get the number of XML items in a sequence.
```

File: examples/xml-functions/xml_functions.bal [hunk 6]
```diff
@@ -31,23 +29,12 @@ public function main() {
...
     // Sets the children elements of an XML element.
     book.setChildren(content);
     io:println(book);
 
-    // Retrieves a particular child of an XML element by its name.
-    x = book.selectDescendants("name");
-    io:println(x);
-
-    // Removes any text items from an XML sequence that are all whitespaces.
+    // Strips the insignificant parts of the an xml value.
```
Comment (praneesha) [inline examples/xml-functions/xml_functions.bal]:
```suggestion
    // Strip the insignificant parts of an XML value.
```

File: examples/xml-functions/xml_functions.bal [hunk 7]
```diff
@@ -31,23 +29,12 @@ public function main() {
...
+    // Comment items, processing instruction items are considered insignificant.
```
Comment (praneesha) [inline examples/xml-functions/xml_functions.bal]:
```suggestion
    // Commented items and processing instruction items are considered insignificant.
```

File: examples/xml-access/xml_access.bal [hunk 8]
```diff
@@ -17,13 +17,13 @@ public function main() {
     // You can access child XML items using xml step expressions.
```
Comment (praneesha) [inline examples/xml-access/xml_access.bal]:
```suggestion
    // You can access child XML items using XML step expressions.
```

File: examples/xml-access/xml_access.bal [hunk 9]
```diff
@@ -17,13 +17,13 @@ public function main() {
...
     io:println(bookXML/<ISBN>/<code>);
 
     // You can also retrieve attributes of the resulting child XML.
     io:println(bookXML/<author>/<fname>.title);
 
-    // You can match descendent elements using following steping access syntax.
+    // You can match descendant elements using the following stepping access syntax.
     io:println(bookXML/**/<fname>);
 
     // Select all children using bellow syntax.
```
Comment (praneesha) [inline examples/xml-access/xml_access.bal]:
```suggestion
    // Select all children using the bellow syntax.
```

File: examples/xml-access/xml_access.bal [hunk 10]
```diff
@@ -32,12 +32,12 @@ public function main() {
     // And select all children elements using bellow syntax.
```
Comment (praneesha) [inline examples/xml-access/xml_access.bal]:
```suggestion
    // Select all children elements using the bellow syntax.
```

File: examples/xml-functions/tests/xml_functions_test.bal [hunk 11]
```diff
@@ -17,29 +17,20 @@ function testFunc() {
     // Invoking the main function
```
Comment (praneesha) [inline examples/xml-functions/tests/xml_functions_test.bal]:
```suggestion
    // Invoking the main function.
```

File: examples/xml-functions/xml_functions.bal [hunk 12]
```diff
@@ -1,27 +1,25 @@
...
     xml someText = xml `Hello, World!`;
     xml content = someText + bookName + bookComment;
 
-    xml book = xml `<book/>`;
-
-    // Gets the type of the XML element.
-    io:println(bookComment.getItemType());
+    xmllib:Element book = <xmllib:Element> xml `<book/>`;
 
     // Gets the name of an XML element.
```
Comment (praneesha) [inline examples/xml-functions/xml_functions.bal]:
```suggestion
    // Get the name of an XML element.
```

File: examples/xml-functions/xml_functions.bal [hunk 13]
```diff
@@ -1,27 +1,25 @@
...
+    xml concat = xmllib:concat(someText, bookName, bookComment);
+    io:println(concat);
+    io:println(content == concat);
 
-    // Checks if the XML element has only one value.
-    io:println(content.isSingleton());
+    // Get the number of xml items in a sequence.
+    io:println(concat.length());
 
     // Gets a subsequence of an XML sequence.
```
Comment (praneesha) [inline examples/xml-functions/xml_functions.bal]:
```suggestion
    // Get a subsequence of an XML sequence.
```

File: examples/xml-functions/xml_functions.bal (hunk 7 above)
Comment (rdhananjaya) [reply examples/xml-functions/xml_functions.bal]: Comment items are a thing in XML ()

Comment (praneesha) [reply examples/xml-functions/xml_functions.bal]: +1

File: examples/xml-access/xml_access.bal [hunk 14]
```diff
@@ -14,30 +14,30 @@ public function main() {
...
 
     // You can also retrieve attributes of the resulting child XML.
     io:println(bookXML/<author>/<fname>.title);
 
-    // You can match descendent elements using following steping access syntax.
+    // You can match descendant elements using the following stepping access syntax.
     io:println(bookXML/**/<fname>);
 
-    // Select all children using bellow syntax.
+    // Select all children using the bellow syntax.
```
Comment (praneesha) [inline examples/xml-access/xml_access.bal]:
```suggestion
    // Select all the children elements using the below syntax.
```

File: examples/xml-access/xml_access.bal [hunk 15]
```diff
@@ -14,30 +14,30 @@ public function main() {
...
     io:println(bookXML/*);
 
-    // And select all children elements using bellow syntax.
+    // Select all children elements using the bellow syntax.
     io:println(bookXML/<*>);
 
-    // Select all children belong to specific namespace.
+    // Select all the children belonging to a specific namespace.
```
Comment (praneesha) [inline examples/xml-access/xml_access.bal]:
```suggestion
    // Select all the children elements belonging to a specific namespace.
```

File: examples/xml-attributes/xml_attributes.description [hunk 16]
```diff
@@ -3,5 +3,5 @@
 // In Ballerina, both of these types are treated the same. 
 // XML attribute access in ballerina is lax typed similar
```
Comment (praneesha) [inline examples/xml-attributes/xml_attributes.description]:
```suggestion
// XML attribute access in Ballerina is lax-typed similar
```

File: examples/xml-attributes/xml_attributes.description [hunk 17]
```diff
@@ -3,5 +3,5 @@
...
 // to `json`, we can use field access expression (`.`) and
-// optional field access expressions (``?.`) to access teh
-// field of a xml value.
+// optional field access expressions (``?.`) to access the
+// attribute of a xml value.
```
Comment (praneesha) [inline examples/xml-attributes/xml_attributes.description]:
```suggestion
// attribute of an XML value.
```

File: examples/xml-functions/xml_functions.bal [hunk 18]
```diff
@@ -1,53 +1,40 @@
...
-
-    // Checks if the XML element has only one value.
-    io:println(content.isSingleton());
-
-    // Gets a subsequence of an XML sequence.
+    // Get a subsequence of an XML sequence.
     xml x = content.slice(2, 3);
     io:println(x);
 
     // Gets all the element-type items in an XML sequence.
```
Comment (praneesha) [inline examples/xml-functions/xml_functions.bal]:
```suggestion
    // Get all the element-type items in an XML sequence.
```

File: examples/xml-functions/xml_functions.bal [hunk 19]
```diff
@@ -1,53 +1,40 @@
...
     x = content.elements();
     io:println(x);
 
-    // Retrieves an XML element by its name.
-    x = content.select("name");
-    io:println(x);
-
     // Sets the children elements of an XML element.
```
Comment (praneesha) [inline examples/xml-functions/xml_functions.bal]:
```suggestion
    // Set the children elements of an XML element.
```

File: examples/xml-access/xml_access.bal (hunk 15 above)
Comment (rdhananjaya) [reply examples/xml-access/xml_access.bal]: Children `elements` are selected using <*> syntax
This select 'elements' and non element items, hence we can omit the `elements` word right?

Comment (praneesha) [reply examples/xml-access/xml_access.bal]: +1

**Pull request merged**